mesa/model.py

```diff
@@ -231,8 +232,19 @@ def reset_rng(self, rng: RNGLike | SeedLike | None = None) -> None:
         Args:
             rng: A new seed for the RNG; if None, reset using the current seed
         """
-        self.rng = np.random.default_rng(rng)
-        self._rng = self.rng.bit_generator.state
+        bit_generators = {
+            entry.__name__: entry
+            for entry in [PCG64, PCG64DXSM, MT19937, Philox, SFC64]
+        }
+
+        if rng is None:
+            rng = self._rng
+            alg = bit_generators[rng["bit_generator"]]()
+            alg.state = rng
+            self.rng = np.random.default_rng(alg)
```

Comment:
Could we do:
```suggestion
        if rng is None:
            # Restore from saved initial state
            bg_class = getattr(np.random, self._rng["bit_generator"])
            bg = bg_class()
            bg.state = self._rng
            self.rng = np.random.Generator(bg)
```
This dynamically retrieves the BitGenerator class name from the saved state dictionary (`self._rng["bit_generator"]`), instantiates it using `getattr(np.random, class_name)`, restores the saved state, and wraps it in a new Generator.